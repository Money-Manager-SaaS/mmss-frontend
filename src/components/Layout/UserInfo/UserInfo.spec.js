import React from 'react';
import renderer from 'react-test-renderer';
import ConnectedUserInfo, { UserInfo } from './UserInfo';
import { MemoryRouter } from 'react-router';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

import configureStore from 'redux-mock-store';


describe('UserInfo render', () => {
  test('pure jest user info snapshot render without redux status', () => {
    const testComp = renderer.create(
      <MemoryRouter>
        <UserInfo email={'a@a.com'} change_auth={() => {
        }}/>
      </MemoryRouter>
    );
    const tree = testComp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('connected user info with provider', () => {
  const initialState = {
    user: {
      email: 'a@a.com',
      change_auth: () => {
      }
    }
  };
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('check email from store', () => {
    wrapper = mount(<Provider store={store}><MemoryRouter><ConnectedUserInfo/></MemoryRouter></Provider>);
    expect(wrapper.find(UserInfo).prop('email')).toEqual(initialState.user.email);
  });
});
