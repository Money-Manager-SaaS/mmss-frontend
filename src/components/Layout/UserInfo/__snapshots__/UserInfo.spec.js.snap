// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo render pure jest user info snapshot render without redux status 1`] = `
<div
  className="user-info"
>
  <a
    className="ant-dropdown-link ant-dropdown-trigger"
    href="/profile"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <article
      className="ant-typography user-name"
    >
      Hi 
      a@a.com
    </article>
     
    <span
      aria-label="down"
      className="anticon anticon-down arrow-down"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="down"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
        />
      </svg>
    </span>
  </a>
  <span
    className="ant-avatar user-symbol ant-avatar-circle"
    style={Object {}}
  >
    <span
      className="ant-avatar-string"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      J
    </span>
  </span>
  <button
    className="ant-btn btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Log Out
    </span>
  </button>
</div>
`;

exports[`UserInfo render user info snapshot render 1`] = `
<div
  className="user-info"
>
  <a
    className="ant-dropdown-link ant-dropdown-trigger"
    href="/profile"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <article
      className="ant-typography user-name"
    >
      Hi 
      a@a.com
    </article>
     
    <span
      aria-label="down"
      className="anticon anticon-down arrow-down"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="down"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
        />
      </svg>
    </span>
  </a>
  <span
    className="ant-avatar user-symbol ant-avatar-circle"
    style={Object {}}
  >
    <span
      className="ant-avatar-string"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      J
    </span>
  </span>
  <button
    className="ant-btn btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Log Out
    </span>
  </button>
</div>
`;
